### PR TITLE
Remove unnecessary attributes of User struct

### PR DESCRIPTION
**Pull request by lunny**
remove unnecessary attributes of User struct

### CODE REVIEW

File: modules/context/repo.go [hunk 1]
```diff
@@ -577,7 +577,8 @@ func RepoAssignment(ctx *Context) (cancel context.CancelFunc) {
 	ctx.Data["BranchName"] = ctx.Repo.BranchName
 
 	// People who have push access or have forked repository can propose a new pull request.
-	canPush := ctx.Repo.CanWrite(unit_model.TypeCode) || (ctx.IsSigned && ctx.User.HasForkedRepo(ctx.Repo.Repository.ID))
+	canPush := ctx.Repo.CanWrite(unit_model.TypeCode) || (ctx.IsSigned &&
+		models.HasForkedRepo(ctx.User.ID, ctx.Repo.Repository.ID))
```
Comment (Gusted) [inline modules/context/repo.go]: Unusual place for a newline to me...
Maybe move the ctx.IsSigned to the next line as well?

File: routers/web/repo/branch.go [hunk 2]
```diff
@@ -54,7 +54,8 @@ func Branches(ctx *context.Context) {
 	ctx.Data["AllowsPulls"] = ctx.Repo.Repository.AllowsPulls()
 	ctx.Data["IsWriter"] = ctx.Repo.CanWrite(unit.TypeCode)
 	ctx.Data["IsMirror"] = ctx.Repo.Repository.IsMirror
-	ctx.Data["CanPull"] = ctx.Repo.CanWrite(unit.TypeCode) || (ctx.IsSigned && ctx.User.HasForkedRepo(ctx.Repo.Repository.ID))
+	ctx.Data["CanPull"] = ctx.Repo.CanWrite(unit.TypeCode) || (ctx.IsSigned &&
```
Comment (Gusted) [inline routers/web/repo/branch.go]: Ditto

File: routers/web/user/home.go [hunk 3]
```diff
@@ -111,7 +111,7 @@ func Dashboard(ctx *context.Context) {
 			return
 		}
 	} else {
-		mirrors, err = ctxUser.GetMirrorRepositories()
+		mirrors, err = models.GetUserMirrorRepositories(ctxUser.ID)
 		if err != nil {
 			ctx.ServerError("GetMirrorRepositories", err)
```
Comment (Gusted) [inline routers/web/user/home.go]:
```suggestion
			ctx.ServerError("GetUserMirrorRepositories", err)
```

File: routers/web/user/profile.go [hunk 4]
```diff
@@ -224,7 +224,7 @@ func Profile(ctx *context.Context) {
 	ctx.Data["Keyword"] = keyword
 	switch tab {
 	case "followers":
-		items, err := ctxUser.GetFollowers(db.ListOptions{
+		items, err := models.GetUserFollowers(ctxUser, db.ListOptions{
```
Comment (Gusted) [inline routers/web/user/profile.go:236]: Line 232:
`GetFollowers` -> `GetUserFollowers`

File: routers/web/user/profile.go [hunk 5]
```diff
@@ -236,7 +236,7 @@ func Profile(ctx *context.Context) {
 
 		total = ctxUser.NumFollowers
 	case "following":
-		items, err := ctxUser.GetFollowing(db.ListOptions{
+		items, err := models.GetUserFollowing(ctxUser, db.ListOptions{
```
Comment (Gusted) [inline routers/web/user/profile.go:248]: As above

File: routers/api/v1/user/follower.go [hunk 6]
```diff
@@ -91,7 +91,7 @@ func ListFollowers(ctx *context.APIContext) {
 }
 
 func listUserFollowing(ctx *context.APIContext, u *models.User) {
-	users, err := u.GetFollowing(utils.GetListOptions(ctx))
+	users, err := models.GetUserFollowing(u, utils.GetListOptions(ctx))
 	if err != nil {
 		ctx.Error(http.StatusInternalServerError, "GetFollowing", err)
```
Comment (Gusted) [inline routers/api/v1/user/follower.go]:
```suggestion
		ctx.Error(http.StatusInternalServerError, "GetUserFollowing", err)
```

File: routers/api/v1/user/follower.go [hunk 7]
```diff
@@ -25,7 +25,7 @@ func responseAPIUsers(ctx *context.APIContext, users []*models.User) {
 }
 
 func listUserFollowers(ctx *context.APIContext, u *models.User) {
-	users, err := u.GetFollowers(utils.GetListOptions(ctx))
+	users, err := models.GetUserFollowers(u, utils.GetListOptions(ctx))
 	if err != nil {
 		ctx.Error(http.StatusInternalServerError, "GetUserFollowers", err)
```
Comment (Gusted) [inline routers/api/v1/user/follower.go:30]: 😄 This was incorrect before.

File: routers/api/v1/repo/pull.go [hunk 8]
```diff
@@ -953,8 +953,8 @@ func parseCompareInfo(ctx *context.APIContext, form api.CreatePullRequestOption)
 	}
 
 	// Check if current user has fork of repository or in the same repository.
-	headRepo, has := models.HasForkedRepo(headUser.ID, baseRepo.ID)
-	if !has && !isSameRepo {
+	headRepo := models.GetForkedRepo(headUser.ID, baseRepo.ID)
+	if headRepo == nil && !isSameRepo {
 		log.Trace("parseCompareInfo[%d]: does not have fork or in same repository", baseRepo.ID)
 		ctx.NotFound("HasForkedRepo")
```
Comment (Gusted) [inline routers/api/v1/repo/pull.go]:
```suggestion
		ctx.NotFound("GetForkedRepo")
```

File: routers/api/v1/org/member.go [hunk 9]
```diff
@@ -190,7 +190,12 @@ func IsPublicMember(ctx *context.APIContext) {
 	if ctx.Written() {
 		return
 	}
-	if userToCheck.IsPublicMember(ctx.Org.Organization.ID) {
+	is, err := models.IsPublicMembership(ctx.Org.Organization.ID, userToCheck.ID)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "IsPublicMember", err)
```
Comment (Gusted) [inline routers/api/v1/org/member.go]:
```suggestion
		ctx.Error(http.StatusInternalServerError, "IsPublicMembership", err)
```

**Review by Gusted — COMMENTED** [review]
Just a few nitpicks otherwise it will L  G TM

**Review by Gusted — APPROVED** [review]
LGTM

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**